### DISCUSSION TITLE
Using bash backslash-escaped prompt characters

### DISCUSSION THREAD

**Discussion by MorganAntonsson** [JanDeDobbeleer/oh-my-posh#2267] [Troubleshoot] [answered]
I am a long time bash user in Linux that finally has decided to try out the new prompt styling tools. Oh My Posh is really nice and very fast! I do miss some things from vanilla bash, though. Especially having the history number and the terminal number in the prompt. With bash, I can achieve this by the backslash-escaped prompt characters '\!' and '\l':
```
export PS1='\! \l $ '
```
Result:
```
1234 0 $ 
```
I thought I could use a text segment to achieve the same result:
```
{
    "$schema":     "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "blocks":      [
        {
            "alignment": "left",
            "segments":  [
                {
                    "foreground": "#fff",
                    "style":      "plain",
                    "template":   "\\! \\l $",
                    "type":       "text"
                }
            ],
            "type":      "prompt"
        }
    ],
    "final_space": true,
    "version":     2
}
```
However, I get an extra backslash in front of each number:
```
\1234 \0 $
```
I figure this is because the template is taken literally instead of the first backslash being interpreted as an escape character. If I change the double backslashes to singles, oh-my-posh fails to parse the theme file:
```
comments:10:35: invalid char escape
comments:10:35: invalid char escape
comments:10:35: invalid char escape
comments:10:35: invalid char escape
Oh My Posh Error:

readEscapedChar: invalid escape char after \, error found in #10 byte of ...|plate":"13402 0 $","ty|..., bigger context ...|"foreground":"#fff","style":"plain","template":"13402 0 $","type":"text"}],"type":"prompt"}],"final_sp|...
```
Does anyone know of a way to get around this?

**Comment by JanDeDobbeleer** [comment, marked as answer]
Yes, that's because we replace `\!` with `\\!` to avoid this from happening (so I'm confused as to why it now does display that). We could agree not to do that for the text segment, and still have this flexibility.

**Reply by MorganAntonsson** [reply]
Thanks for the quick reply! Yeah, that would be great! Do you want me to write a ticket?

**Reply by JanDeDobbeleer** [reply]
@MorganAntonsson almost there my friend

```json
{
    "foreground": "#fff",
    "style": "powerline",
    "background": "p:blue",
    "powerline_symbol": "\ue0b0",
    "interactive": true,
    "template": " \\! \\l ",
    "type": "text"
}
```

<img width="1170" alt="image" src="https://user-images.githubusercontent.com/2492783/168444806-1207f3b8-1628-4837-af2d-7faa64e8a13d.png">

**Reply by JanDeDobbeleer** [reply]
@MorganAntonsson when [this](https://github.com/JanDeDobbeleer/oh-my-posh/pull/2268) lands, you'll have these capabilities. Just need to double check something.

**Reply by MorganAntonsson** [reply]
> @MorganAntonsson when [this](https://github.com/JanDeDobbeleer/oh-my-posh/pull/2268) lands, you'll have these capabilities. Just need to double check something.

That's awesome! Thanks a lot!
